### PR TITLE
Create metric card for Frugal Score

### PR DESCRIPTION
**Pull request by sashavor**
Proposing metric card for Frugal Score.

@albertvillanova or @lhoestq  -- there are certain aspects that I'm not 100% sure on (such as how exactly the distillation between BertScore and FrugalScore is done) -- so if you find that something isn't clear, please let me know!

### CODE REVIEW

File: metrics/frugalscore/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# Metric Card for FrugalScore
+
+
+## Metric Description
+FrugalScore is a reference-based metric for NLG models evaluation. It is based on a distillation approach that allows to learn a fixed, low cost version of any expensive NLG metric, while retaining most of its original performance.
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:5]: Nit
```suggestion
FrugalScore is a reference-based metric for Natural Language Generatio (NLG) models evaluation. It is based on a distillation approach that allows to learn a fixed, low cost version of any expensive NLG metric, while retaining most of its original performance.
```

File: metrics/frugalscore/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+## How to use 
+
+When loading FrugalScore, you can indicate the model you wish to use to compute the score. The default model is `tiny_bert-base_bert-score`, and a full list of models can be found in the [Limitations and bias](#Limitations-and-bias) section.
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:9]:
```suggestion
When loading FrugalScore, you can indicate the model you wish to use to compute the score. The default model is `moussaKam/frugalscore_tiny_bert-base_bert-score`, and a full list of models can be found in the [Limitations and bias](#Limitations-and-bias) section.
```

File: metrics/frugalscore/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+```python
+>>> from datasets import load_metric
+>>> frugalscore = load_metric("frugalscore", checkpoint = "moussaKam/frugalscore_medium_bert-base_mover-score")
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:13]: Bug: if you pass `checkpoint` arg, this is ignored. I have tried your code and the metric's model and tokenizer were using the default `'moussaKam/frugalscore_tiny_bert-base_bert-score'`:
```python
In [2]: frugalscore = load_metric("frugalscore", checkpoint="moussaKam/frugalscore_medium_bert-base_mover-score")
In [3]: frugalscore.tokenizer
Out[3]: PreTrainedTokenizerFast(name_or_path='moussaKam/frugalscore_tiny_bert-base_bert-score', vocab_size=30522, model_max_len=1000000000000000019884624838656, is_fast=True, padding_side='right', special_tokens={'unk_token': '[UNK]', 'sep_token': '[SEP]', 'pad_token': '[PAD]', 'cls_token': '[CLS]', 'mask_token': '[MASK]'})
```

```suggestion
>>> frugalscore = load_metric("frugalscore", "moussaKam/frugalscore_medium_bert-base_mover-score")
```

```python
In [6]: frugalscore = load_metric("frugalscore", "moussaKam/frugalscore_medium_bert-base_mover-score")

In [7]: frugalscore.tokenizer
Out[7]: PreTrainedTokenizerFast(name_or_path='moussaKam/frugalscore_medium_bert-base_mover-score', vocab_size=30522, model_max_len=1000000000000000019884624838656, is_fast=True, padding_side='right', special_tokens={'unk_token': '[UNK]', 'sep_token': '[SEP]', 'pad_token': '[PAD]', 'cls_token': '[CLS]', 'mask_token': '[MASK]'})

File: metrics/frugalscore/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+FrugalScore calculates how good are the predictions given some references, based on a set of scores.
+
+The inputs it takes are:
+
+`predictions` : a list of strings representing the predictions to score.
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:20]: Nit
```suggestion
`predictions`: a list of strings representing the predictions to score. 
```

File: metrics/frugalscore/README.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+>>> frugalscore = load_metric("frugalscore", checkpoint = "moussaKam/frugalscore_medium_bert-base_mover-score")
+```
+
+FrugalScore calculates how good are the predictions given some references, based on a set of scores.
+
+The inputs it takes are:
+
+`predictions` : a list of strings representing the predictions to score. 
+
+`references` : a list of string representing the references for each prediction.
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:22]:
```suggestion
`references`: a list of string representing the references for each prediction. 
```

File: metrics/frugalscore/README.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+The inputs it takes are:
+
+`predictions` : a list of strings representing the predictions to score. 
+
+`references` : a list of string representing the references for each prediction. 
+
+Its optional arguments are:
+
+`batch_size` : the batch size for predictions (default value is `32`).
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:26]:
```suggestion
`batch_size`: the batch size for predictions (default value is `32`).
```

File: metrics/frugalscore/README.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+`max_length`: the maximum sequence length (default value is `128`).
+
+`device`: either gpu or cpu (default value is `None`).
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:30]:
```suggestion
`device`: either "gpu" or "cpu" (default value is `None`). 
```

File: metrics/frugalscore/README.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+`device`: either gpu or cpu (default value is `None`). 
+
+```python
+>>> results = frugalscore.compute(predictions=['hello there', 'huggingface'], references=['hello world', 'hugging face'], batch_size = 16, max_length = 64, device = "gpu")
+```
+
+## Output values
+
+The output of FrugalScore is a dictionary with the list of scores:
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:38]: Suggestion?
```suggestion
The output of FrugalScore is a dictionary with the list of scores for each prediction-reference pair:
```

File: metrics/frugalscore/README.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+Its optional arguments are:
+
+`batch_size` : the batch size for predictions (default value is `32`).
+
+`max_length`: the maximum sequence length (default value is `128`).
+
+`device`: either gpu or cpu (default value is `None`). 
+
+```python
+>>> results = frugalscore.compute(predictions=['hello there', 'huggingface'], references=['hello world', 'hugging face'], batch_size = 16, max_length = 64, device = "gpu")
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:33]: Style:
```suggestion
>>> results = frugalscore.compute(predictions=['hello there', 'huggingface'], references=['hello world', 'hugging face'], batch_size=16, max_length=64, device="gpu")
```

File: metrics/frugalscore/README.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+| [small_roberta_bert-score](https://huggingface.co/moussaKam/frugalscore_small_roberta_bert-score)    | BERT-small       | RoBERTa-Large | BERTScore |
+| [medium_roberta_bert-score](https://huggingface.co/moussaKam/frugalscore_medium_roberta_bert-score)   | BERT-medium  | 	RoBERTa-Large| BERTScore |
+| [tiny_deberta_bert-score](https://huggingface.co/moussaKam/frugalscore_tiny_deberta_bert-score)   | BERT-tiny  | DeBERTa-XLarge| BERTScore |
+| [small_deberta_bert-score](https://huggingface.co/moussaKam/frugalscore_small_deberta_bert-score)   | BERT-small  | DeBERTa-XLarge | BERTScore |
+| [medium_deberta_bert-score](https://huggingface.co/moussaKam/frugalscore_medium_deberta_bert-score)   | BERT-medium  | DeBERTa-XLarge | BERTScore |
+| [tiny_bert-base_mover-score](https://huggingface.co/moussaKam/frugalscore_tiny_bert-base_mover-score)   | BERT-tiny | BERT-Base | MoverScore |
+| [small_bert-base_mover-score](https://huggingface.co/moussaKam/frugalscore_small_bert-base_mover-score)  | BERT-small  | BERT-Base	 | MoverScore |
+| [medium_bert-base_mover-score](https://huggingface.co/moussaKam/frugalscore_medium_bert-base_mover-score)   | BERT-medium  | BERT-Base | MoverScore |
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:87]: Model names should be complete, so that it can be properly identified and downloaded, e.g. `moussaKam/frugalscore_tiny_bert-base_bert-score` instead of `tiny_bert-base_bert-score`
```suggestion
| FrugalScore                                        | Student     | Teacher        | Method     |
|----------------------------------------------------|-------------|----------------|------------|
| [moussaKam/frugalscore_tiny_bert-base_bert-score](https://huggingface.co/moussaKam/frugalscore_tiny_bert-base_bert-score)    | BERT-tiny   | BERT-Base      | BERTScore  |
| [moussaKam/frugalscore_small_bert-base_bert-score](https://huggingface.co/moussaKam/frugalscore_small_bert-base_bert-score)   | BERT-small  | BERT-Base      | BERTScore  |
| [moussaKam/frugalscore_medium_bert-base_bert-score](https://huggingface.co/moussaKam/frugalscore_medium_bert-base_bert-score) | BERT-medium | BERT-Base      | BERTScore  |
| [moussaKam/frugalscore_tiny_roberta_bert-score](https://huggingface.co/moussaKam/frugalscore_tiny_roberta_bert-score)     | BERT-tiny   | RoBERTa-Large  | BERTScore  |
| [moussaKam/frugalscore_small_roberta_bert-score](https://huggingface.co/moussaKam/frugalscore_small_roberta_bert-score)     | BERT-small  | RoBERTa-Large  | BERTScore  |
| [moussaKam/frugalscore_medium_roberta_bert-score](https://huggingface.co/moussaKam/frugalscore_medium_roberta_bert-score)    | BERT-medium | RoBERTa-Large  | BERTScore  |
| [moussaKam/frugalscore_tiny_deberta_bert-score](https://huggingface.co/moussaKam/frugalscore_tiny_deberta_bert-score)      | BERT-tiny   | DeBERTa-XLarge | BERTScore  |
| [moussaKam/frugalscore_small_deberta_bert-score](https://huggingface.co/moussaKam/frugalscore_small_deberta_bert-score)     | BERT-small  | DeBERTa-XLarge | BERTScore  |
| [moussaKam/frugalscore_medium_deberta_bert-score](https://huggingface.co/moussaKam/frugalscore_medium_deberta_bert-score)    | BERT-medium | DeBERTa-XLarge | BERTScore  |
| [moussaKam/frugalscore_tiny_bert-base_mover-score](https://huggingface.co/moussaKam/frugalscore_tiny_bert-base_mover-score)   | BERT-tiny   | BERT-Base      | MoverScore |
| [moussaKam/frugalscore_small_bert-base_mover-score](https://huggingface.co/moussaKam/frugalscore_small_bert-base_mover-score)  | BERT-small  | BERT-Base      | MoverScore |
| [moussaKam/frugalscore_medium_bert-base_mover-score](https://huggingface.co/moussaKam/frugalscore_medium_bert-base_mover-score) | BERT-medium | BERT-Base      | MoverScore |
```

**Review by albertvillanova — CHANGES_REQUESTED** [review]
Thanks again for adding this. Just small corrections below.

For the details about the specific distillation approach, I don't know either. Maybe Quentin can give more insight.

File: metrics/frugalscore/README.md (hunk 3 above)
Comment (sashavor) [reply metrics/frugalscore/README.md:13]: haha so, do we fix the bug, or leave as is?

File: metrics/frugalscore/README.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
+# Metric Card for FrugalScore
+
+
+## Metric Description
+FrugalScore is a reference-based metric for NLG models evaluation. It is based on a distillation approach that allows to learn a fixed, low cost version of any expensive NLG metric, while retaining most of its original performance.
+
+The FrugalScore models are obtained by continuing the pretraining of small models on a synthetic dataset constructed using summarization, backtranslation and denoising models. During the training, the small models learn the internal mapping of the expensive metric, including any similarity function.
+
+The FrugalScore models are obtained by continuing the pretraining of small models on a synthetic dataset constructed using summarization, backtranslation and denoising models. During the training, the small models learn the internal mapping of the expensive metric, including any similarity function.
```
Comment (albertvillanova) [inline metrics/frugalscore/README.md:9]: Repeated...

File: metrics/frugalscore/README.md (hunk 1 above)
Comment (albertvillanova) [reply metrics/frugalscore/README.md:5]: And this was missed

Comment (sashavor) [reply metrics/frugalscore/README.md:5]: fixed in the new PR!
